frame/elections-phragmen/src/lib.rs

```diff
@@ -627,36 +510,206 @@ decl_module! {
-				0
-			}
+		/// Note that old members and runners-up are also candidates.
+		CandidateSlashed(<T as frame_system::Config>::AccountId, BalanceOf<T>),
+		/// A \[seat holder\] was slashed by \[amount\] by being forcefully removed from the set.
+		SeatHolderSlashed(<T as frame_system::Config>::AccountId, BalanceOf<T>),
+	}
+
+	#[deprecated(note = "use `Event` instead")]
+	pub type RawEvent<T> = Event<T>;
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Cannot vote when no candidates or members exist.
+		UnableToVote,
+		/// Must vote for at least one candidate.
+		NoVotes,
+		/// Cannot vote more than candidates.
+		TooManyVotes,
+		/// Cannot vote more than maximum allowed.
+		MaximumVotesExceeded,
+		/// Cannot vote with stake less than minimum balance.
+		LowBalance,
+		/// Voter can not pay voting bond.
+		UnableToPayBond,
+		/// Must be a voter.
+		MustBeVoter,
+		/// Cannot report self.
+		ReportSelf,
+		/// Duplicated candidate submission.
+		DuplicatedCandidate,
+		/// Member cannot re-submit candidacy.
+		MemberSubmit,
+		/// Runner cannot re-submit candidacy.
+		RunnerUpSubmit,
+		/// Candidate does not have enough funds.
+		InsufficientCandidateFunds,
+		/// Not a member.
+		NotMember,
+		/// The provided count of number of candidates is incorrect.
+		InvalidWitnessData,
+		/// The provided count of number of votes is incorrect.
+		InvalidVoteCount,
+		/// The renouncing origin presented a wrong `Renouncing` parameter.
+		InvalidRenouncing,
+		/// Prediction regarding replacement after member removal is wrong.
+		InvalidReplacement,
+	}
+
+	#[pallet::origin]
+	pub struct Origin<T>(PhantomData<T>);
+
+	/// The current elected members.
+	///
+	/// Invariant: Always sorted based on account id.
+	#[pallet::storage]
+	#[pallet::getter(fn members)]
+	pub type Members<T: Config> =
+		StorageValue<_, Vec<SeatHolder<T::AccountId, BalanceOf<T>>>, ValueQuery>;
+
+	/// The current reserved runners-up.
+	///
+	/// Invariant: Always sorted based on rank (worse to best). Upon removal of a member, the
+	/// last (i.e. _best_) runner-up will be replaced.
+	#[pallet::storage]
+	#[pallet::getter(fn runners_up)]
+	pub type RunnersUp<T: Config> =
+		StorageValue<_, Vec<SeatHolder<T::AccountId, BalanceOf<T>>>, ValueQuery>;
+
+	/// The present candidate list. A current member or runner-up can never enter this vector
+	/// and is always implicitly assumed to be a candidate.
+	///
+	/// Second element is the deposit.
+	///
+	/// Invariant: Always sorted based on account id.
+	#[pallet::storage]
+	#[pallet::getter(fn candidates)]
+	pub type Candidates<T: Config> = StorageValue<_, Vec<(T::AccountId, BalanceOf<T>)>, ValueQuery>;
+
+	#[pallet::type_value]
+	pub fn DefaultForElectionRounds() -> u32 {
+		Zero::zero()
+	}
+
+	/// The total number of vote rounds that have happened, excluding the upcoming one.
+	#[pallet::storage]
+	#[pallet::getter(fn election_rounds)]
+	pub type ElectionRounds<T: Config> = StorageValue<_, u32, ValueQuery, DefaultForElectionRounds>;
```

Comment:
```suggestion
	/// The total number of vote rounds that have happened, excluding the upcoming one.
	#[pallet::storage]
	#[pallet::getter(fn election_rounds)]
	pub type ElectionRounds<T: Config> = StorageValue<_, u32, ValueQuery>;
```

zero is the default for u32